src/app/containers/ComscoreAnalytics/Canonical/index.jsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable jsx-a11y/alt-text */
+import React, { useContext, useEffect, useState } from 'react';
+import { Helmet } from 'react-helmet';
+import { UserContext } from '#contexts/UserContext';
+import onClient from '#lib/utilities/onClient';
+
+const CanonicalComscoreAnalytics = () => {
+  const { personalisationEnabled } = useContext(UserContext);
+  const [comscoreConsent, setComscoreConsent] = useState('0');
+
+  const staticAssetsPath = `${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN}${process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH}`;
+
+  useEffect(() => {
+    const csUcfr = personalisationEnabled ? '1' : '0';
+    /* eslint-disable no-underscore-dangle */
+    window._comscore = window._comscore || [];
+    window._comscore.push({ c1: '2', c2: '17986528', cs_ucfr: csUcfr });
+    setComscoreConsent(csUcfr);
+  }, [personalisationEnabled]);
+
+  // We do not want to return the comscore script on the server
+  if (!onClient()) {
```

Comment:
Hey, sorry I'm just looking at this again and it doesn't seem quite right. We need to render the `<noscript>` tag at least on the server - if there's no javascript enabled, we won't execute this code on the client, so the noscript tag is a bit pointless. I don't understand how we're supposed to read the cookie value on the client and pass it to this image URL though, that's impossible if we're not running javascript.

So, I think what makes sense is:

- Return everything on the server and client
- Keep this useEffect to set the values in the `_comscore` variable
- Replace `cs_ucfr=${comscoreConsent}` with `cs_ucfr=0` in the image tag, and remove comscoreConsent from state - we can't update this on the client if JS is disabled.

Might want to double check noscript behaviour with Jim, or Jon in Jim's absence.

Reply:
I've updated it in this commit: https://github.com/bbc/simorgh/pull/7391/commits/49d59235f59bf4cd347fd5d4f50f77906508b300
so that the canonical comscore scripts are rendered on both client and server.